### PR TITLE
[camerax] Add `LifecycleOwner` Proxy

### PR DESCRIPTION
**Pull request by camsim99**
Adds proxy implementation of `LifecycleOwner` for `Activity`s this plugin may be bound to.

Heavily inspired by [`google_maps_flutter_android`](https://github.com/flutter/packages/blob/main/packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapsPlugin.java#L121).

Fixes https://github.com/flutter/flutter/issues/125695.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera_android_camerax/test/camera_test.dart [hunk 1]
```diff
@@ -5,10 +5,18 @@
...
 
+import 'camera_test.mocks.dart';
+import 'test_camerax_library.g.dart';
+
+@GenerateMocks(<Type>[TestInstanceManagerHostApi])
 void main() {
   TestWidgetsFlutterBinding.ensureInitialized();
 
+  // Mocks the call to clear the native InstanceManager.
+  TestInstanceManagerHostApi.setup(MockTestInstanceManagerHostApi());
```
Comment (camsim99) [inline packages/camera/camera_android_camerax/test/camera_test.dart:18]: Unrelated fix for failing Dart test.

File: packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java
Comment (reidbaker) [inline packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java:1]: I think you should add a test for this file to make sure the fallback works.

File: packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ProxyLifecycleProvider.java
Comment (reidbaker) [inline packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ProxyLifecycleProvider.java:1]: Non blocking: Considering this is at least the second plugin to need this style code consider either including the class in the framework or publishing it as its own package.

**Review by reidbaker — APPROVED** [review]

File: packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java [hunk 2]
```diff
@@ -79,9 +80,17 @@ public void onAttachedToActivity(@NonNull ActivityPluginBinding activityPluginBi
         pluginBinding.getApplicationContext(),
         pluginBinding.getTextureRegistry());
     updateContext(pluginBinding.getApplicationContext());
-    processCameraProviderHostApi.setLifecycleOwner(
-        (LifecycleOwner) activityPluginBinding.getActivity());
-    systemServicesHostApi.setActivity(activityPluginBinding.getActivity());
+
+    Activity activity = activityPluginBinding.getActivity();
+
+    if (activity instanceof LifecycleOwner) {
```
Comment (bparrishMines) [inline packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java:86]: Is the `Activity` in the engine not always a `LifecycleOwner`?

**Review by bparrishMines — APPROVED** [review]
LGTM

File: packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java (hunk 2 above)
Comment (bparrishMines) [reply packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java:86]: Actually, now that I think about it. This probably isn't guaranteed in an Add-to-app scenario.

Comment (camsim99) [reply packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraAndroidCameraxPlugin.java:86]: Yeah I think that's why the cast wasn't safe.

File: packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ProxyLifecycleProvider.java
Comment (camsim99) [reply packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/ProxyLifecycleProvider.java:1]: Will include this consideration in my umbrella issue for plugin cleanup.

**Pull request merged**